pkg/relayer/config/relayminer_configs_reader.go

```diff
@@ -1,102 +1,68 @@
 	}
 
-	// Parse the query node URL
-	queryNodeUrl, err := url.Parse(yamlRelayMinerConfig.QueryNodeUrl)
-	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidQueryNodeUrl.Wrapf("%s", err)
-	}
-
-	// Check that the network node URL is provided
-	if yamlRelayMinerConfig.NetworkNodeUrl == "" {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("network node url is required")
-	}
-
-	// Parse the network node URL
-	networkNodeUrl, err := url.Parse(yamlRelayMinerConfig.NetworkNodeUrl)
-	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("%s", err)
-	}
-
-	// Parse the websocket URL of the Pocket Node to connect to for subscribing to on-chain events.
-	pocketNodeWebsocketUrl := sdk.HostToWebsocketURL(queryNodeUrl.Host)
-
-	if yamlRelayMinerConfig.SigningKeyName == "" {
+	// Top level section
+	// SigningKeyName is required
+	if len(yamlRelayMinerConfig.SigningKeyName) == 0 {
 		return nil, ErrRelayMinerConfigInvalidSigningKeyName
 	}
+	relayMinerConfig.SigningKeyName = yamlRelayMinerConfig.SigningKeyName
 
-	if yamlRelayMinerConfig.SmtStorePath == "" {
+	// SmtStorePath is required
+	if len(yamlRelayMinerConfig.SmtStorePath) == 0 {
 		return nil, ErrRelayMinerConfigInvalidSmtStorePath
 	}
+	relayMinerConfig.SmtStorePath = yamlRelayMinerConfig.SmtStorePath
 
-	if yamlRelayMinerConfig.ProxiedServiceEndpoints == nil {
-		return nil, ErrRelayMinerConfigInvalidServiceEndpoint.Wrapf("proxied service endpoints are required")
+	// Hydrate the pocket node urls
+	if err := relayMinerConfig.HydratePocketNodeUrls(&yamlRelayMinerConfig.PocketNode); err != nil {
+		return nil, err
 	}
 
-	if len(yamlRelayMinerConfig.ProxiedServiceEndpoints) == 0 {
-		return nil, ErrRelayMinerConfigInvalidServiceEndpoint.Wrapf("no proxied service endpoints provided")
+	// Hydrate the proxies
+	if err := relayMinerConfig.HydrateProxies(yamlRelayMinerConfig.Proxies); err != nil {
+		return nil, err
 	}
 
-	// Parse the proxied service endpoints
-	proxiedServiceEndpoints := make(map[string]*url.URL, len(yamlRelayMinerConfig.ProxiedServiceEndpoints))
-	for serviceId, endpointUrl := range yamlRelayMinerConfig.ProxiedServiceEndpoints {
-		endpoint, err := url.Parse(endpointUrl)
-		if err != nil {
-			return nil, ErrRelayMinerConfigInvalidServiceEndpoint.Wrapf("%s", err)
-		}
-		proxiedServiceEndpoints[serviceId] = endpoint
+	// Hydrate the suppliers
+	if err := relayMinerConfig.HydrateSuppliers(yamlRelayMinerConfig.Suppliers); err != nil {
+		return nil, err
 	}
 
-	relayMinerCMDConfig := &RelayMinerConfig{
-		QueryNodeUrl:            queryNodeUrl,
-		NetworkNodeUrl:          networkNodeUrl,
-		PocketNodeWebsocketUrl:  pocketNodeWebsocketUrl,
-		SigningKeyName:          yamlRelayMinerConfig.SigningKeyName,
-		ProxiedServiceEndpoints: proxiedServiceEndpoints,
-		SmtStorePath:            yamlRelayMinerConfig.SmtStorePath,
+	// Check if a proxy is referencing a host more than once
+	for _, proxyConfig := range relayMinerConfig.Proxies {
```

Comment:
Can you add a `Validate` function (with the appropriate receiver as the config) so this is just a one liner that returns an error?